packages/material-ui/src/Drawer/Drawer.test.js

```diff
@@ -41,183 +45,152 @@ describe('<Drawer />', () => {
+
+        setProps({ open: true });
 
-        wrapper.setProps({ open: true });
-        wrapper.update();
-        expect(wrapper.find(Slide).props().in).to.equal(true);
+        expect(screen.getByTestId('child')).not.to.equal(null);
 
-        wrapper.setProps({ open: false });
-        wrapper.update();
-        expect(wrapper.find(Slide).props().in).to.equal(false);
+        setProps({ open: false });
+        act(() => {
+          clock.tick(transitionDuration);
+        });
+
+        expect(screen.queryByTestId('child')).to.equal(null);
       });
     });
   });
 
   describe('prop: variant=persistent', () => {
-    const drawerElement = (
-      <Drawer variant="persistent">
-        <h1>Hello</h1>
-      </Drawer>
-    );
-
     it('should render a div instead of a Modal when persistent', () => {
-      const wrapper = mount(drawerElement);
-      const root = findOutermostIntrinsic(wrapper);
-      expect(root.type()).to.equal('div');
-      expect(root.hasClass(classes.docked)).to.equal(true);
+      const { container } = render(
+        <Drawer variant="persistent">
+          <div />
+        </Drawer>,
+      );
+
+      expect(container.firstChild).to.have.tagName('div');
+      expect(container.firstChild).to.have.class(classes.docked);
     });
 
     it('should render Slide > Paper inside the div', () => {
-      const wrapper = mount(drawerElement);
-      const div = wrapper.find('div').first();
-      const slide = div.childAt(0);
-      expect(slide.length).to.equal(1);
-      expect(slide.type()).to.equal(Slide);
-
-      const paper = findOutermostIntrinsic(slide);
-      expect(paper.exists()).to.equal(true);
-      expect(paper.hasClass(classes.paper)).to.equal(true);
+      const transitionDuration = 123;
+      const handleEntered = spy();
+      const { container, setProps } = render(
+        <Drawer
+          open={false}
+          transitionDuration={transitionDuration}
+          SlideProps={{ onEntered: handleEntered }}
+          variant="persistent"
+        >
+          <div />
+        </Drawer>,
+      );
+
+      setProps({ open: true });
+
+      expect(handleEntered.callCount).to.equal(0);
+
+      act(() => {
+        clock.tick(transitionDuration);
+      });
+
+      expect(handleEntered.callCount).to.equal(1);
+      expect(container.firstChild.firstChild).to.have.class(classes.paper);
     });
   });
 
   describe('prop: variant=permanent', () => {
     const drawerElement = (
       <Drawer variant="permanent">
-        <h1>Hello</h1>
+        <div />
       </Drawer>
     );
 
     it('should render a div instead of a Modal when permanent', () => {
       const { container } = render(drawerElement);
       const root = container.querySelector(`.${classes.root}`);
+
+      expect(root).not.to.equal(null);
       expect(root).to.have.tagName('div');
       expect(root).to.have.class(classes.docked);
     });
-
-    it('should render div > Paper inside the div', () => {
```

Comment:
Moved the core assertion to "hould render a div instead of a Modal when permanent"